pybossa/cache/projects.py

```diff
@@ -121,10 +125,13 @@ def format_task(row, lock_users=[]):
     if filter_user_prefs:
         # construct task list for worker view
         params["assign_user"] = args["sql_params"]["assign_user"]
-        results = session.execute(text(sql), params)
         task_rank_info = []
-
         user_profile = args.get("filter_by_wfilter_upref", {}).get("current_user_profile", {})
+        results = []
+        with session.begin_nested():
+            session.execute("SET LOCAL enable_indexscan = OFF;")
+            results = session.execute(text(sql), params)
```

Comment:
Does it need a session.commit() to indicate a transaction ends? Or probably the "with" statement does the auto commit by implementing the logic in the __enter()__ method in the session? Just would like to double check.

Reply:
TBH I didn't find useful resources about whether we need session.commit(), I am assuming the "with" statement does this. However, I am sure that the "SET LOCAL .." does not apply to the session, because when I refresh the page, it always starts with "OFF" value.

Reply:
It will be committed when the with context exit (```__exit()__``` method). See the [code](https://github.com/zzzeek/sqlalchemy/blob/81518ae2e2bc622f8cd47287a575ad4c0e43ead1/lib/sqlalchemy/orm/session.py#L558-L569). So I think we are good.

Reply:
Hold on.... looks like the new version might just close the session at the end of "with": https://github.com/zzzeek/sqlalchemy/blob/master/lib/sqlalchemy/orm/session.py#L1135